packages/server-wallet/src/db/migrations/20201014174039_open-channel-objectives.ts

```diff
@@ -0,0 +1,17 @@
+import * as Knex from 'knex';
+
+const tableName = 'open-channel-objectives';
+
+export async function up(knex: Knex): Promise<any> {
+  return knex.schema.createTable(tableName, function(table) {
+    table.integer('objectiveId');
+    table.string('status');
+    table.string('type');
+    table.string('targetChannelId');
```

Comment:
This should be a foreign key to the `channels` table

Reply:
Most likely yes, as long as the objective is never inserted before the target channel (which could feasibly happen during opening a channel). It could be a nullable foreign key.

Reply:
That sounds like a client bug to me. What happens if the client crashes before inserting the channel? What happens if the client decides to delete a closed channel but forgot to delete the associated objective?